workflow/controller/operator.go

```diff
@@ -350,22 +351,28 @@ func (woc *wfOperationCtx) persistUpdates() {
 	if !woc.updated {
 		return
 	}
-	wf := woc.wf.DeepCopy()
-	wfClient := woc.controller.wfclientset.ArgoprojV1alpha1().Workflows(wf.ObjectMeta.Namespace)
+	wfClient := woc.controller.wfclientset.ArgoprojV1alpha1().Workflows(woc.wf.ObjectMeta.Namespace)
 	// try and compress nodes if needed
-	nodes := wf.Status.Nodes
-	err := packer.CompressWorkflow(wf)
-	if packer.IsTooLargeError(err) {
-		wf.Status.OffloadNodeStatus = woc.controller.offloadNodeStatusRepo.IsEnabled()
+	nodes := woc.wf.Status.Nodes
+
+	err := packer.CompressWorkflow(woc.wf)
+	if packer.IsTooLargeError(err) || os.Getenv("ALWAYS_OFFLOAD_NODE_STATUS") == "true" {
```

Comment:
Instead of introducing ALWAYS_OFFLOAD_NODE_STATUS, for e2e tests, we should leverage the existing env var, MAX_WORKFLOW_SIZE and set it to 0. This will specifically exercise the `packer.CompressWorkflow()` logic and achieve the same effect you are looking for.

Reply:
I had originally done this, but it did not do what I wanted, which was always offload.

Reply:
As discussed - we'll keep this.